### PR TITLE
Doc: Add comparison to related libs + document replace

### PR DESCRIPTION
**Pull request by jatcwang**

### CODE REVIEW

File: docs/docs/docs/BestPracticeAndFAQ.md [hunk 1]
```diff
@@ -39,3 +39,24 @@ Fully automatic derivation is intentionally left out for compile time reasons.
...
+
+Feature-wise, difflicious has:
+
+- Better collection diffing: Difflicious allows you to specify how Seq/Set elements are paired for comparison. 
+  Pairing also allows you compare Seqs order-independently.
+- Better configurability: Difflicious takes a more "structured" approach to configurability, where Differ of a complex type
+  can still have all its underlying Differs tweaked or even replaced (using `replace`). This is handy in some scenarios
+  where you can reuse existing Differs by "swapping" them in and out of a larger Differ.
+
+DiffX also has fully automatic derivation on by default, which can lead to long compile times.
```
Comment (jatcwang) [inline docs/docs/docs/BestPracticeAndFAQ.md:59]: @ghostbuster91 do you think this is a fair comparison?

Comment (ghostbuster91) [reply docs/docs/docs/BestPracticeAndFAQ.md:59]: Hi, thanks for letting me know about that.

> DiffX also has fully automatic derivation on by default, which can lead to long compile times.

Diffx supports both automatic and semi-automatic derivation, but in order to use automatic derivation one has to explicitly turn if on either by importing or extending a particular trait. Use of the automatic derivation has been discouraged in the readme for some time already (due to the compilation times). 

After examining your library I think that two major differences from the user perspective are:
- diffx supports both automatic and semi-automatic derivation
- diffx main typeclass is contravariant in `T` while difficious is invariant
- diffx support scalajs

Another significant difference is that diffx typeclasses are opaque(? I am not sure if this is the correct term), i.e. they cannot be inspected at runtime, while difficious can be. This although, doesn't matter that much to end users. However that directly corresponds to the times when configuration happens. In difficious because of that approach configuration happens on the definition site, and so the potential errors will be thrown there. In diffx, on the other hand, configuration does nothing until the comparison is invoked, so errors are thrown on the call site. Anyway, this doesn't matter that much because the configuration is usually done through macro application, so mostly it should be correct.

> Better collection diffing: Difflicious allows you to specify how Seq/Set elements are paired for comparison.

There is an `ObjectMatcher` for that purpose in diffx. It might not be as handy as difficious, but recently I pushed some changes to improve its ergonomics. But for now, yes.

>  Pairing also allows you compare Seqs order-independently.

Like above, this feature is almost ready.

> Better configurability: Difflicious takes a ...

TLDR, yes, for now :) 

This is something I wanted to do for a while in diffx, but there was a lot of uncertain things. 
I have seen two possible approaches: 
- generalizing diffx `ignore` mechanism (introducing DiffContext which will holds overrided diff instances)
- creating more structured typeclasses as you did

Personally I like your approach more and I think that it could have been implemented into diffx, but as you already did it, I am going to use the first approach (see this pr for details https://github.com/softwaremill/diffx/pull/261 )


Last but not least I wonder if these two libraries are really so different in their approaches :)

Comment (jatcwang) [reply docs/docs/docs/BestPracticeAndFAQ.md:59]: Thanks for all the writeup. I'll definitely take a look at diffx again since my knowledge is definitely a few months stale :)
I started working on difflicious because I didn't want to push for major breaking changes (e.g. no auto-derivation) or features that requires major internal overhaul. 

If our overall philosophy and feature set isn't so much different I'm definitely in favor of combining efforts, even if that means I have to say goodbye to the best library name I've ever came up with ;)

File: docs/docs/docs/QuickStart.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+```
+"com.github.jatcwang" %% "difflicious-munit" % "{{ site.version }}" % Test
+```
+
+If you are running tests using **IntelliJ IDEA**'s test runner, you will want 
+to turn off the red text coloring it uses for test failure outputs because
+it interferes with difflicious' color outputs.
+
+In <b>File | Settings | Editor | Color Scheme | Console Colors | Console | Error Output</b>, uncheck the red foreground color.
```
Comment (jatcwang) [inline docs/docs/docs/QuickStart.md:21]: @ghostbuster91 in diffx's readme you mentioned issues with intellij's output coloring. I fixed it this way if it helps

File: docs/docs/docs/BestPracticeAndFAQ.md (hunk 1 above)
Comment (jatcwang) [reply docs/docs/docs/BestPracticeAndFAQ.md]: I'll go ahead and merge this PR with a disclaimer that the comparison may be out of date :)

Comment (ghostbuster91) [reply docs/docs/docs/BestPracticeAndFAQ.md]: Sure, thanks :) Sorry for the delay, but I am currently on vacation.

File: docs/docs/docs/QuickStart.md (hunk 2 above)
Comment (ghostbuster91) [reply docs/docs/docs/QuickStart.md:21]: Awesome, thanks for the tip!
